internal/changelog/renderer.go

```diff
@@ -124,11 +126,11 @@ func (r Renderer) Render() error {
 	return afero.WriteFile(r.fs, outFile, data.Bytes(), changelogFilePerm)
 }
 
-//go:embed asciidoc-template.asciidoc
+//go:embed *.asciidoc
 var asciidocTemplate embed.FS
 
 func (r Renderer) Template() ([]byte, error) {
-	data, err := asciidocTemplate.ReadFile("asciidoc-template.asciidoc")
+	data, err := asciidocTemplate.ReadFile(r.templ)
```

Comment:
I like the solution but would add a slight change to the way we manage the default.

I would use a more descriptive string, like `asciidoc-embedded`, and add an if in this method to check for it. If the value of `templ` is equal to our string the function returns the embedded template. In this way we don't need to embed `*.asciidoc`, which may created unwanted side effects, but we can just embed `asciidoc-template.asciidoc` as before using it's value when needed.
What do you think?

Reply:
Are you suggesting renaming `asciidoc-template` to `asciidoc-embedded`?

Also, should the custom template provided by the user be read from the file if it was not the default embedded template?

Reply:
Also, I think it might make sense to add a `templates` folder to the project, so the user doesn't actually need to dive into `/internal/changelog` where the embedded template is found to use a custom one. What do you think @endorama?

Reply:
These questions feel like there is a misunderstanding going on :)

Let me try to clarify the context: If a developer wants to add a new template it can embed it in the software, but if a user wants to use a template that is not embedded they can't. The CLI will have an embedded template (maybe more than one in the future, but one today) that can be triggered by using a magic string. We want users to be able to supply at runtime (so no embedding possible) an arbitrary file to be used as a template.

> Are you suggesting renaming asciidoc-template to asciidoc-embedded?

I'm suggesting to use `asciidoc-embedded` as magic value to trigger the usage of the embedded template, instead of the name of the file (which is something users will know nothing about).

> Also, I think it makes sense to add a templates folder to the project,

Yes but let's name it `assets`, as the template file is an asset this CLI uses.

Let me know if it's clearer now, otherwise let's catch up with a brief call to discuss this.

Reply:
Yes, I definitely understand your suggestion better.

Right now:
- `internal/changelog` -> developers embed templates, users can access them in config/flag via `magic strings`. 
- `assets/` -> users create their own templates, accessing them via the file names.

Those magic strings should be specified somewhere in the docs.

What do you think @endorama?

Reply:
> assets/ -> users create their own templates, accessing them via the file names.

As it is possible to directly use the provided flag value as file path, what is the advantage of using the `assets` folder? (I though we would use it to store embedded templates to help with discoverability for contributors)

Reply:
My only concern with this is that you can't directly embed from the `assets/` path in `renderer.go`: https://github.com/golang/go/issues/46056

One solution is to create a package inside the `assets` folder and import that into  `renderer.go`.
Moving `getEmbeddedTemplates` there as well for consistency.

Reply:
I forgot about that detail around embedding.

> One solution is to create a package inside the assets folder and import embedded templates from there using a variable into renderer.go.
> This package can also contain getEmbeddedTemplates as it makes more sense for the magic string map to be there.

This is interesting but to avoid confusion the package should be `internal/assets`.
I'm ok moving forward with `internal/assets` if you think it make sense and is an improvement.
Otherwise let's just keep it like it is now, your choice 😊

Reply:
Updated the code and docs a little bit. 

Now the available embedded templates can be listed using the `--help` flag in the `render` command which will help users.

Let me know 😄